.github/workflows/build.yaml

```diff
@@ -0,0 +1,268 @@
+          BRANCH_SLUG: ${{ steps.extract_branch.outputs.BRANCH_SLUG }}
+        shell: bash
+        run: |
+          # determine version
+          GIT_LAST_TAG=${{ steps.get_previous_tag.outputs.tag }}
+          GIT_NEXT_TAG=${{ steps.get_next_semver_tag.outputs.patch }}
+          echo "GIT_LAST_TAG=${GIT_LAST_TAG}, GIT_NEXT_TAG=${GIT_NEXT_TAG}"
+
+          if [[ "$BRANCH" == "release-"* ]]; then
+            # Release Branch: extract version from branch name
+            VERSION=${BRANCH#"release-"}
+            else
+            if [[ "$BRANCH" == "master" ]]; then
+              # master branch = latest-dev
+              VERSION="${GIT_NEXT_TAG}-dev"
+            else
+              # Feature/Development Branch - use last tag with branch slug
+              VERSION="${GIT_NEXT_TAG}-dev-${BRANCH_SLUG}"
+            fi
+          fi
+
+          echo "VERSION=${VERSION}"
+
+          echo "##[set-output name=VERSION;]$(echo ${VERSION})"
+      - name: Get current date and time
+        id: get_datetime
+        run: |
+          echo "::set-output name=DATE::$(date +'%Y%m%d')"
+          echo "::set-output name=TIME::$(date +'%H%M')"
+          echo "::set-output name=DATETIME::$(date +'%Y%m%d')$(date +'%H%M')"
+
+  ############################################################################
+  # Build Checks                                                             #
+  ############################################################################
+  build_checks:
+    needs: [prepare_ci_run]
+    strategy:
+      matrix:
+        component: [ "ci-connect-cli", "promotion-service", "git-operator" ]
+
+    runs-on: ubuntu-20.04
+    env:
+      BRANCH: ${{ needs.prepare_ci_run.outputs.BRANCH }}
+      VERSION: ${{ needs.prepare_ci_run.outputs.VERSION }}
+      DATETIME: ${{ needs.prepare_ci_run.outputs.DATE }}${{ needs.prepare_ci_run.outputs.TIME }}
+      GIT_SHA: ${{ needs.prepare_ci_run.outputs.GIT_SHA }}
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.29
+          working-directory: ${{ matrix.component }}
+          args: --timeout 120s
+
+      - name: Test
+        working-directory: ${{ matrix.component }}
+        run: go test ./...
+
+  ############################################################################
+  # Build Container Images                                                   #
+  ############################################################################
+  container_build:
+    needs: [prepare_ci_run, build_checks]
+    strategy:
+      matrix:
+        component: [ "ci-connect-cli", "promotion-service", "git-operator" ]
```

Comment:
This workflow apparently builds all components already, I guess the workflows from other the other PRs can be dropped.